### PR TITLE
Add Board and Game class, game logic and playability.

### PR DESCRIPTION
**Pull request by pointerish**
This PR implements the following:

- A Board class representing the board state and functionality and containing a #win? method that checks whether the last move was winning
- General game logic and playability.

### CODE REVIEW

**Comment by mimipeshy** [comment]
## STATUS : CHANGES REQUIRED :recycle:  :red_circle: 

Hello @pointerish :wave:  :wave: this is really good work you did, it's impressive :smile: However we still need to fix a few more things  before approval

- [ ] No inline comments left after the previous milestone. Board.rb has commented code, remove them
##### Behaviour

* [ ] The application does not allow a player to move into an occupied position. The user should get an error when they try this move.
* [ ] The application does not exist/stop due to invalid input - it informs the player about it and asks for their move again.
* [ ] The application can determine if the move is a winning move. This is caused by hard coding the number of moves both players can make, so it exits if the limit is reached.
`9.times do |turn| # There are only 9 possible moves until draw`
 It can be done with a while loop. Check for the following issues too:
  * [ ] The application can detect a sequence of three horizontal elements that are equal. 
  * [ ] The application can detect a sequence of three vertical elements that are equal
  * [ ] The application can detect a sequence of three equal elements in a diagonal from left and right to right
  * [ ] The application can tell if there was a draw after players move

#### Implementation details
* [ ] The project uses at least two classes (Usually it uses three: Game, Board, and Player). The code of the project should be logically divided between classes. Move the game logic from main.rb to its own class and just initialize the game in main.rb.
* [ ] All game-logic related resources (classes and/or modules) are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)). Move board, game and player to lib folder.
* [ ] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). Students tends to create long methods that are doing multiple steps of the game process. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/single_responsibility.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

  - Refactor main.rb file to achieve this


Thats all :relieved:  now give me a smile :smile: 
Submit another review form when done.

Regards 
Peris Ndanu
Feel free to slack me if you have any questions.

**Comment by pointerish** [comment]
Hello @mimipeshy,

Thanks for the feedback. Makes sense. The only thing confusing here, at least for me, is the way everything is written. 

> The application does not allow a player to move into an occupied position. The user should get an error when they try this move.

This reads as if the application is in fact not allowing the invalid move when I guess it means the opposite given the second statement. The same happens for all of the bullet points and it's a tad confusing. 

Thanks for all the help!

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Board
+  attr_accessor :board
+
+  def initialize
+    # [[-, -, -],
+    #  [-, -, -],
+    #  [-, -, -]]
```
Comment (descholar-ceo) [inline lib/board.rb:7]: Remove this commented codes

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Change Required :vertical_traffic_light: :warning: 
## Hello @pointerish  :handshake: :handshake: :handshake: 
:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
## Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes]
- [x] No inline comments left after the previous milestone. Board.rb has commented code, remove them
- [ ] The game shouldn't stop if there is a win or a draw, it should ask players if they would like to play another round, and then they press [Y/N],
- [x] When there is draw, the game is exiting before upating the board with the last move, fix that as well
Screenshot from 2020-11-18 09-23-34


If anything seems to be unclear to you, don't hesitate to contact me for more clarification slack: `@Mugirase Emmanuel` | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)

**Review by PhillipUg — APPROVED** [review]
# Status: Project Approved! 🥇 
Congratulations on making all the requested changes and getting this project approved. :+1:

You can merge this branch into the development branch when you're ready and go on to your next project.
:bulb:  Do not forget to submit your project completion form after merging this branch.

All the best team,

For any assistance, you can contact me on slack as **Phillip Musiime**

**Pull request merged**